lib/pairmotron/pair_maker.ex

```diff
@@ -47,8 +47,12 @@ defmodule Pairmotron.PairMaker do
 
     determination = PairBuilder.determify(pairs, users)
 
-    determination.dead_pairs
-      |> Enum.map(fn(p) -> Repo.delete! p end)
+    multi = Ecto.Multi.new
+
+    multi = determination.dead_pairs
+      |> Enum.reduce(multi, fn(pair, acc) ->
+          acc |> Ecto.Multi.delete(:remove_dead_pairs, pair)
```

Comment:
You might be able to use Multi.delete_all as well, but I think you might have to construct a query there. You might gain some clarity. This looks fine though.